oauthproxy.go

```diff
@@ -504,11 +541,28 @@ func (p *OAuthProxy) IsValidRedirect(redirect string) bool {
 		if err != nil {
 			return false
 		}
+		redirectHostname := redirectURL.Hostname()
+
 		for _, domain := range p.whitelistDomains {
-			if (redirectURL.Host == domain) || (strings.HasPrefix(domain, ".") && strings.HasSuffix(redirectURL.Host, domain)) {
-				return true
+			domainHostname, domainPort := splitHostPort(strings.TrimLeft(domain, "."))
+			if err != nil || domainHostname == "" {
```

Comment:
We have no `err` around this part of the code, doesn't seem to make sense to check it here right?

Reply:
You're right, this seems to be left over from old code. Good catch